### PR TITLE
PIN-4293 Updated the use of fingerprint

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala [hunk 1]
```diff
@@ -72,21 +68,17 @@ object Utils {
...
+    getOpenApiExtractedMetrics(fingerprint) orElse getWSDLExtractedMetrics(fingerprint)
 
-  private def getOpenApiExtractedMetrics(bytes: Array[Byte]): Either[Throwable, FileExtractedMetrics] =
+  private def getOpenApiExtractedMetrics(fingerprint: String): Either[Throwable, FileExtractedMetrics] =
     InterfaceParser
-      .parseOpenApi(bytes)
-      .flatMap(json =>
-        InterfaceParserUtils.getEndpoints(json).map(es => FileExtractedMetrics(Digester.toSha256(bytes), es.size))
-      )
+      .parseOpenApi(fingerprint.getBytes())
```
Comment (beetlecrunch) [inline padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala]: Here we are parsing the interface, so this implementation it's not correct

File: padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala [hunk 2]
```diff
@@ -72,21 +68,17 @@ object Utils {
...
+      .flatMap(json => InterfaceParserUtils.getEndpoints(json).map(es => FileExtractedMetrics(fingerprint, es.size)))
 
-  private def getWSDLExtractedMetrics(bytes: Array[Byte]): Either[Throwable, FileExtractedMetrics] =
+  private def getWSDLExtractedMetrics(fingerprint: String): Either[Throwable, FileExtractedMetrics] =
     InterfaceParser
-      .parseWSDL(bytes)
-      .flatMap(xml =>
-        InterfaceParserUtils.getEndpoints(xml).map(es => FileExtractedMetrics(Digester.toSha256(bytes), es.size))
-      )
+      .parseWSDL(fingerprint.getBytes())
```
Comment (beetlecrunch) [inline padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala]: same

File: padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala (hunk 1 above)
Comment (nttdata-rtorsoli) [reply padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala]: I need further explanation

File: padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala (hunk 2 above)
Comment (nttdata-rtorsoli) [reply padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala]: same

File: padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala (hunk 1 above)
Comment (nttdata-rtorsoli) [reply padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala]: Reverted waiting for how to proceed

File: padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala (hunk 2 above)
Comment (nttdata-rtorsoli) [reply padigitale-report-generator/src/main/scala/it/pagopa/interop/padigitalereportgenerator/util/Utils.scala]: same

**Comment by beetlecrunch** [comment]
This improvement will be implemented in the TS version.

**Pull request closed without merging**